plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ExpressionConverter.java

```diff
@@ -84,6 +85,9 @@ public static Expression toIcebergExpression(TupleDomain<IcebergColumnHandle> tu
         List<Expression> conjuncts = new ArrayList<>();
         for (Map.Entry<IcebergColumnHandle, Domain> entry : domainMap.entrySet()) {
             IcebergColumnHandle columnHandle = entry.getKey();
+            if (isMetadataColumnId(columnHandle.getId())) {
+                continue;
```

Comment:
Let's move this logic to the place which applies conditions on metadata columns, so that we know no constraint gets ignored.

thus, here it would be `checkArgument(! isMetadataColumnId(columnHandle.getId()))`
and `IcebergSplitSource` would divide `TupleDomain` into 

1. $path domain --> handled directly there
2. the reset --> passed to `toIcebergExpression`